### PR TITLE
Database deployments

### PR DESCRIPTION
**Pull request by OctopusBob**
Please review and let me know if I need to change anything!

### CODE REVIEW

File: docs/deployment-examples/sql-server-databases.md [hunk 1]
```diff
@@ -1,48 +1,91 @@
...
+
+2. A change-driven approach, where schema changes are manually written and only run once.  The target database keeps track of which scripts already ran.
+
+[Redgate's ReadyRoll](https://www.red-gate.com/products/sql-development/readyroll/index), [DbUp](https://dbup.readthedocs.io/en/latest/), and [RoundhousE](http://projectroundhouse.org/) use the change-driven approach.
+
+There are pros and cons to either approach as well as the tools themselves.  It is important for your team to research the tools and determine the best one for you.
+
+## Tentacle Installation Recommendations {#SQLServerdatabases-Tentacles}
+
+Deploying an IIS Web Application or a Windows Service is very straight-forward.  Install the tentacle on the server to be deployed to.  SQL Server is different.  Production SQL Servers are typically clusters or high-availability groups.  They comprise more than one node hidden behind a VIP or virtual IP Address.
```
Comment (leecarlon) [inline docs/deployment-examples/sql-server-databases.md]: Tentacle should be capitalized throughout.

File: docs/deployment-examples/sql-server-databases.md [hunk 2]
```diff
@@ -1,48 +1,91 @@
...
+
+![](common-database-with-vip.png "width=500")
+
+For high-availability groups, there is an active node and a passive node.  In this case, installing a tentacle on each node will not work.  Octopus Deploy will see multiple tentacles and attempt to deploy to both nodes.
+
+SQL PaaS, such as [AWS RDS](https://aws.amazon.com/rds/) or [Azure's SQL](https://azure.microsoft.com/en-us/services/sql-database/), will not allow the installation of tentacles on SQL Server.  
+
+All the tools mentioned above connect to SQL Server using port 1433 and run one or more scripts.  They do not need to be installed directly on SQL Server.  Nor do they need to be run directly on SQL Server.  They will work as long as they run on any machine with a direct connection and port 1433 open.
+
+Also, windows authentication is the often the preferred way to authenticate.  A recommended security practice is the principle of least privilege.  The account used by the website to connect to SQL Server should have little permissions.  Whereas, the account used to make schema changes needs elevated permissions.
```
Comment (leecarlon) [inline docs/deployment-examples/sql-server-databases.md]: Hi Bob, should "have little permissions" be something like "have the fewest possible permissions"?

File: docs/deployment-examples/sql-server-databases.md [hunk 3]
```diff
@@ -1,48 +1,91 @@
...
+
+Most of the tooling from above requires it to be installed somewhere.  It is important the same version is used across all environments.  This prevents any uncertainty during deployments.  
+
+Finally, it is good security practice to have a different account with schema change permissions per environment.  An account used to change a test environment should not be able to change production.
```
Comment (leecarlon) [inline docs/deployment-examples/sql-server-databases.md]: It took me a little while to properly parse: 
Finally, it is good security practice to have a different account with schema change permissions per environment.
Perhaps something like:
Finally, it is good security practice to use different accounts (with schema change permissions) per environment.

File: docs/deployment-examples/sql-server-databases.md [hunk 4]
```diff
@@ -1,48 +1,91 @@
...
+
+Finally, it is good security practice to have a different account with schema change permissions per environment.  An account used to change a test environment should not be able to change production.  
+
+With all that in mind, a "jump box" is where tentacles should be installed.  The jump box sits between Octopus Deploy and the SQL Server VIP.  The tentacle is running as a [service account](/docs/infrastructure/windows-targets/running-tentacle-under-a-specific-user-account.md) with the necessary permissions to make schema changes.  The tooling chosen for database deployments is installed on the jump box.
+
+![](database-with-jump-box.png "width=500") 
+
+In the event of multiple domains, a jumpbox would be needed per domain.  This might be seen where there is a domain in a local infrastructure and another domain in a cloud provider such as Azure.  As long as port 10933 is open (for a listening tentacle) or port 443 (for a polling tentacle) Octopus will be able to communicate to the jumpbox.
```
Comment (leecarlon) [inline docs/deployment-examples/sql-server-databases.md]: Minor point: "a jumpbox would be needed per domain" -> "a jump box is needed per domain".

File: docs/deployment-examples/sql-server-databases.md [hunk 5]
```diff
@@ -1,48 +1,91 @@
...
+
+### Deployment Permission Considerations {#SQLServerdatabases-DeploymentPermissions}
+
+The account used to make schema changes requires elevated permissions.  Because of that, create a special account to handle database deployments.  Do not use the same account used by an IIS Web Application.
+
+The level of elevated permissions is up to you.   The more restrictions placed on the deployment account means more manual steps.  Deployments will fail due to missing or restricted permissions.  Octopus will provide the error message to fix the issue.  It will need a manual intervention.  It is up to you to decide which is best. 
+
+First, decide what the deployment account should have the ability to do at the server level.  From there, research which server roles can are applicable.  For example, the account can create databases and users.  Use the securityadmin and dbcreator server roles.  Should the account only be able to create databases? Use dbcreator role.
```
Comment (leecarlon) [inline docs/deployment-examples/sql-server-databases.md]: First, decide what the deployment account should have the ability to do at the server level.  From there, research which server roles ~~can~~ are applicable.

File: docs/deployment-examples/sql-server-databases.md (hunk 1 above)
Comment (OctopusBob) [reply docs/deployment-examples/sql-server-databases.md]: Thank you, fixed.

File: docs/deployment-examples/sql-server-databases.md (hunk 2 above)
Comment (OctopusBob) [reply docs/deployment-examples/sql-server-databases.md]: I've cleaned it up to read "should have restricted permissions" and provided an example.

File: docs/deployment-examples/sql-server-databases.md (hunk 5 above)
Comment (OctopusBob) [reply docs/deployment-examples/sql-server-databases.md]: I'm surprised both HemingwayApp and Grammarly didn't catch that.  Good find.

**Pull request merged**